Added indentation options to `JSON3.pretty(...)`

jlbosse opened the pull request:
I added a struct `AlignmentContext` that is passed to `JSON3.pretty(...)` which specifies the indentation type (left- or colon-aligned), the indentation level and spaces per indentation level.

This fixes #161

Please let know if this looks good and if you want me to add a sentence or two to the documentation about this.

quinnj commented on src/pretty.jl:
```diff
@@ -8,23 +8,51 @@ macro pretty(json)
+    offset::UInt16 = 0
+    function AlignmentContext(alignment, indent, level, offset)
+        if alignment != :Left && alignment != :Colon
+            throw(ArgumentError("Alignment :$(alignment) is not supported. \
```
I think the trailing `\` character on this line is causing some tests to fail

jlbosse replied:
I changed that line to something more 1.0 friendly. Hopefully the tests pass now!

quinnj commented:
Sorry for the slow response here; this LGTM! I left one comment on something to fix so the 1.0 tests pass, but then we're good to merge!

arnaudh commented:
Don't know if this the right place, but could this MR also fix this minor issue I've just opened? https://github.com/quinnj/JSON3.jl/issues/225
(Don't know if this would be fixed with the new alignment logic)

jlbosse commented:
It wasn't fixed before with the new alignment logic, but should be now.

quinnj commented:
@jlbosse, it looks like the docs are failing here; my guess is that the `Base.kwdef` is messing that up somehow? I think if you just moved the docs after the struct definition like:
```julia
"""
[docs]
"""
AlignmentContext
```
that should probably work

jlbosse commented:
Reading the error messsages it seems like it is complaining that `AlignmentContext` is mentioned nowhere in the docs. I changed that and added an example how to use it to switch between left aligning and  aligning at the colon.

jlbosse commented:
Re the failing code coverage tests: I did add tests for different alignment types. So I am not quite sure what it is complaining about and get a 403 error when trying to view the report (presumably because I am not a project maintainer).

quinnj commented:
I'm not too worried about the coverage since the patch shows an increase.

quinnj commented:
Thanks @jlbosse !

Wikunia commented:
Would love to use this feature as well. Is there anything blocking a release of a new version?

quinnj commented:
Oops, forgot about the new release; we were waiting to get one more PR in and that got merged the other day. Initiated the new release here: https://github.com/quinnj/JSON3.jl/commit/3d542d32bc5d9fddce2feef7ce577db626e0195c

Wikunia commented:
Awesome thanks @quinnj and @jlbosse looking forward to using it
